### PR TITLE
[Manual backport 2.x] Use smaller and compressed varients of buttons and form components (#2068)

### PR DESCRIPTION
**Pull request by AMoo-Miki**
cherry-picked 69d12cee1b0e43869d078e30cc996e9d685e44a9 from #2068

Note: There were a ton of conflicts due to PRs that have not been backported.

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by TackAdam — APPROVED** [review]

**Comment by derek-ho** [comment]
Looks like some tests are failing. @ps48 can we make sure backport failed PRs are backported as expected?

**Comment by ps48** [comment]
@derek-ho I usually try to keep an eye on PRs expected to be backported. But at the end it is a joint responsibility of the PR authors, maintainers (all 14) and the reviewers together to make sure that the PR is backported as expected. So, let's not blame/point individuals here. 

As of now there are 3 major PRs that keep main and 2.x out of sync for dashboards-observability.

PR1: https://github.com/opensearch-project/dashboards-observability/pull/1828 (This should be backported)
PR2: https://github.com/opensearch-project/dashboards-observability/pull/1871 (This should be reverted as the changes here are not going to be shipped in near future) 
PR3: https://github.com/opensearch-project/dashboards-observability/pull/1834  (This should be reverted as the changes here are not going to be shipped in near future)

**Comment by TackAdam** [comment]
It looks like the only failing test on  the mac build is from the query assist:
FAIL public/components/event_analytics/explorer/query_assist/__tests__/input.test.tsx (17.189 s)
I thought this was a flaky test?

File: public/components/common/search/search.tsx [hunk 1]
```diff
@@ -6,9 +6,10 @@
 import '@algolia/autocomplete-theme-classic';
 import {
   EuiBadge,
+  EuiSmallButton,
```
Comment (ps48) [inline public/components/common/search/search.tsx:9]: Should we replace the existing button import here?

File: public/components/custom_panels/panel_modules/visualization_container/visualization_container.tsx [hunk 2]
```diff
@@ -5,7 +5,8 @@
 
 import {
   EuiButton,
-  EuiButtonIcon,
+  EuiSmallButton,
```
Comment (ps48) [inline public/components/custom_panels/panel_modules/visualization_container/visualization_container.tsx:8]: Same here. Not sure if EuiButton replacement was missed here or this one is expected to not be changed

File: public/components/event_analytics/explorer/events_views/surrounding_flyout.tsx [hunk 3]
```diff
@@ -6,15 +6,16 @@
...
+  EuiSmallButtonEmpty,
   EuiButtonIcon,
   EuiCallOut,
-  EuiFieldNumber,
+  EuiCompressedFieldNumber,
   EuiFlexGroup,
   EuiFlexItem,
   EuiFlyoutBody,
   EuiFlyoutHeader,
+  EuiCompressedFormRow,
```
Comment (ps48) [inline public/components/event_analytics/explorer/events_views/surrounding_flyout.tsx:18]: Might have missed a form row here.

File: public/components/trace_analytics/components/dashboard/latency_trend_cell.tsx [hunk 4]
```diff
@@ -3,8 +3,14 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { EuiButtonIcon, EuiFlexGroup, EuiFlexItem, EuiPopover, EuiText } from '@elastic/eui';
-import _ from 'lodash';
+import {
+  EuiButtonIcon,
```
Comment (ps48) [inline public/components/trace_analytics/components/dashboard/latency_trend_cell.tsx:7]: Might have missed the Euibuttonicon here

File: public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/logs_view_config_panel_item.tsx [hunk 5]
```diff
@@ -6,11 +6,12 @@
 import React, { useEffect, useState, useContext, Fragment } from 'react';
 import {
   EuiTitle,
-  EuiComboBox,
+  EuiCompressedComboBox,
   EuiSpacer,
+  EuiSmallButton,
```
Comment (ps48) [inline public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/logs_view_config_panel_item.tsx:11]: might have missed a button replacement here?

**Review by ps48 — APPROVED** [review]
Nothing blocking except the comments above. @AMoo-Miki Please let us know if you update the comments in this PR or want to create an issue and fix them later.

**Review by sumukhswamy — APPROVED** [review]

File: public/components/common/search/search.tsx (hunk 1 above)
Comment (AMoo-Miki) [reply public/components/common/search/search.tsx:9]: No. there are `EuiButton` that were already `size="s"`.

File: public/components/custom_panels/panel_modules/visualization_container/visualization_container.tsx (hunk 2 above)
Comment (AMoo-Miki) [reply public/components/custom_panels/panel_modules/visualization_container/visualization_container.tsx:8]: No. there are `EuiButton` instances that were already `size="s"`.

File: public/components/event_analytics/explorer/events_views/surrounding_flyout.tsx (hunk 3 above)
Comment (AMoo-Miki) [reply public/components/event_analytics/explorer/events_views/surrounding_flyout.tsx:18]: No. There is a `EuiFormRow` that has its own `display`.

File: public/components/trace_analytics/components/dashboard/latency_trend_cell.tsx (hunk 4 above)
Comment (AMoo-Miki) [reply public/components/trace_analytics/components/dashboard/latency_trend_cell.tsx:7]: The `EuiButtonIcon` already has a `size`.

File: public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/logs_view_config_panel_item.tsx (hunk 5 above)
Comment (AMoo-Miki) [reply public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/logs_view_config_panel_item.tsx:11]: There is an `EuiButton` with a `size` on this page as well.

**Comment by AMoo-Miki** [comment]
> Nothing blocking except the comments above. @AMoo-Miki Please let us know if you update the comments in this PR or want to create an issue and fix them later.

Thanks for reviewing Pratik. Those imports are because the files already had instances with different `size` or `display` props that we do not want to override.

**Pull request merged**